keyboards/keebio/tragicforce68/keymaps/rossman360/keymap.c

```diff
@@ -0,0 +1,237 @@
+  LAUNCHCLOCK = SAFE_RANGE,
+  CLOCKIN,
+  CLOCKOUT,
+  CSPEAK,
+  SPEAK1,
+  SPEAK2,
+  SPEAK3,
+  SPEAK4,
+  PARADOWN,
+  WREFRESH,
+  SMERGE,
+  PMERGE,
+  URIGHT,
+  ULEFT,
+  REMCAPS,
+  COPYCLOSE
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+   case CSPEAK:
+      if (record->event.pressed) {
+        SEND_STRING(SS_TAP(X_PGDOWN));
+        SEND_STRING(SS_TAP(X_ENTER));
+        SEND_STRING(SS_TAP(X_ENTER));
+SEND_STRING(SS_TAP(X_PGDOWN));
+ break;
+ }
+ case SPEAK1:
+      if (record->event.pressed) {
+        SEND_STRING(SS_TAP(X_PGDOWN));
+        SEND_STRING(SS_TAP(X_ENTER));
+        SEND_STRING(SS_TAP(X_ENTER));
+SEND_STRING(SS_TAP(X_PGDOWN));
+SEND_STRING(SS_LCTRL(SS_TAP(X_1)));
+}
+break;
+case SPEAK2:
+      if (record->event.pressed) {
+        SEND_STRING(SS_TAP(X_PGDOWN));
+        SEND_STRING(SS_TAP(X_ENTER));
+        SEND_STRING(SS_TAP(X_ENTER));
+SEND_STRING(SS_TAP(X_PGDOWN));
+SEND_STRING(SS_LCTRL(SS_TAP(X_2)));
+}
+break;
+case SPEAK3:
+      if (record->event.pressed) {
+        SEND_STRING(SS_TAP(X_PGDOWN));
+        SEND_STRING(SS_TAP(X_ENTER));
+        SEND_STRING(SS_TAP(X_ENTER));
+SEND_STRING(SS_TAP(X_PGDOWN));
+SEND_STRING(SS_LCTRL(SS_TAP(X_3)));
+}
+break;
+case SPEAK4:
+      if (record->event.pressed) {
+        SEND_STRING(SS_TAP(X_PGDOWN));
+        SEND_STRING(SS_TAP(X_ENTER));
+        SEND_STRING(SS_TAP(X_ENTER));
+SEND_STRING(SS_TAP(X_PGDOWN));
+SEND_STRING(SS_LCTRL(SS_TAP(X_4)));
+}
+break;
+case PARADOWN:
+      if (record->event.pressed) {
+ SEND_STRING(SS_TAP(X_PGDOWN));
+ SEND_STRING(SS_TAP(X_ENTER));
+ SEND_STRING(SS_TAP(X_PGDOWN));
+      }
+      break;
+      case ULEFT:
+      if (record->event.pressed) {
+      SEND_STRING(SS_TAP(X_UP));
+      SEND_STRING(SS_TAP(X_HOME));
+      }
+      break;
+      case URIGHT:
+      if (record->event.pressed) {
+      SEND_STRING(SS_TAP(X_UP));
+SEND_STRING(SS_TAP(X_END));      
+      }
+      break;
+      case SMERGE:
+      if (record->event.pressed) {
+        SEND_STRING(SS_TAP(X_DELETE));
+SEND_STRING(SS_LCTRL(SS_TAP(X_BSPACE)));
+SEND_STRING(SS_LCTRL(SS_TAP(X_BSPACE)));
+      }
+      break;
+      case PMERGE:
+      if (record->event.pressed) {
+      SEND_STRING(SS_TAP(X_HOME));
+SEND_STRING(SS_TAP(X_BSPACE));
+SEND_STRING(SS_TAP(X_SPACE));
+SEND_STRING(SS_LCTRL(SS_TAP(X_BSPACE)));
+SEND_STRING(SS_TAP(X_SPACE));
+      }
+      break;
+      case WREFRESH:
+      if (record->event.pressed) {
+ SEND_STRING(SS_TAP(X_SPACE));
+ SEND_STRING(SS_TAP(X_BSPACE));
+      }
+      break;
+      case REMCAPS:
+      if (record->event.pressed) {
+ SEND_STRING(SS_TAP(X_LEFT));
+ SEND_STRING(SS_TAP(X_LEFT));
+ SEND_STRING(SS_LCTRL(SS_TAP(X_LEFT)));
+ SEND_STRING(SS_TAP(X_DELETE));
+      }
+      break;
+      case COPYCLOSE:
+   if (record->event.pressed) {
+          SEND_STRING(SS_LCTRL(SS_TAP(X_C)));
+          SEND_STRING(SS_LCTRL(SS_TAP(X_W)));
+      }
+      break;
+    };  
+  return true;
+  
+};
```

Comment:
The indentation here could use some love, but functionally this looks ok.  My only other suggestion would be to convert all these multi-line SEND_STRINGs into a single line like so, unless you are doing this out of a desire for readability:

```
SEND_STRING(SS_TAP(X_PGDOWN));
SEND_STRING(SS_TAP(X_ENTER));
SEND_STRING(SS_TAP(X_ENTER));
SEND_STRING(SS_TAP(X_PGDOWN));
```

Could be condensed to:

```
SEND_STRING(SS_TAP(X_PGDOWN) SS_TAP(X_ENTER) SS_TAP(X_ENTER) SS_TAP(X_PGDOWN));
```

Reply:
love it.